File: java/colors/colors.java
public class Main {
    final static int FOREGROUND = 38;
    final static int BACKGROUND = 48;
    final static int INITIAL = 0;
    final static int FINAL = 255;

    static void color_range(int start, int end, int anyground) {
        String resetdefault = "\u001b[0m";
        String resetwithspace = resetdefault.concat(" ");
        String resetoutput = resetwithspace;

        for (int color = start; color <= end; color++) {
            if (color % 16 == 0) {
                System.out.println();
            }

            if (color == end) {
                resetoutput = resetdefault;
            }

            System.out.printf("\u001b[%d;5;%dm%3d%s", anyground, color, color, resetoutput);
        }
    }

    public static void main(String[] args) {
        System.out.printf("\u001b[1;39m%d to %d Colors (FOREGROUND)\u001b[22;0m\n", INITIAL, FINAL);
        color_range(INITIAL, FINAL, FOREGROUND);
        System.out.println("\n");

        System.out.printf("\u001b[1;39m0 to 255 Colors (BACKGROUND)\u001b[22;0m\n", INITIAL, FINAL);
        color_range(INITIAL, FINAL, BACKGROUND);
        System.out.println("\n");
    }
}

File: perl/colors/colors.pl
use v5.16;

my ($FOREGROUND, $BACKGROUND, $INITIAL, $FINAL) = (38, 48, 0, 255);

my $resetdefault = "\x1b[0m";
my $resetwithspace = "$resetdefault ";

sub color_range {
    my ($start, $end, $anyground) = @_;
    my $resetoutput = $resetwithspace;

    for (my $color = $start; $color <= $end; $color++) {
        # Add a newline every 16 columns with the help of `say` function.
        say if ($color % 16 == 0);

        if ($color == $end) {
            $resetoutput = $resetdefault;
        }

        printf("\x1b[%d;5;%dm%3d%s", $anyground, $color, $color, $resetoutput);
    }
}

printf("\x1b[1;37m%d to %d Colors (FOREGROUND)%s\n", $INITIAL, $FINAL, $resetdefault);
color_range($INITIAL, $FINAL, $FOREGROUND);
say "\n";

printf("\x1b[1;37m%d to %d Colors (BACKGROUND)%s\n", $INITIAL, $FINAL, $resetdefault);
color_range($INITIAL, $FINAL, $BACKGROUND);
say "\n";
